backends/candle/src/models/debertav2.rs

```diff
@@ -0,0 +1,1490 @@
+where
+    D: Deserializer<'de>,
+{
+    #[derive(Deserialize, Debug)]
+    #[serde(untagged)]
+    enum StringOrVec {
+        String(String),
+        Vec(Vec<String>),
+    }
+
+    match StringOrVec::deserialize(deserializer)? {
+        StringOrVec::String(s) => Ok(s.split('|').map(String::from).collect()),
+        StringOrVec::Vec(v) => Ok(v),
+    }
+}
+
+// https://github.com/huggingface/transformers/blob/78b2929c0554b79e0489b451ce4ece14d265ead2/src/transformers/models/deberta_v2/modeling_deberta_v2.py#L823
+pub struct DebertaV2Embeddings {
+    device: Device,
+    word_embeddings: Embedding,
+    position_embeddings: Option<Embedding>,
+    token_type_embeddings: Option<Embedding>,
+    layer_norm: LayerNorm,
+    position_ids: Tensor,
+    config: DebertaV2Config,
+    embedding_size: usize,
+    embed_proj: Option<candle_nn::Linear>,
+}
+
+impl DebertaV2Embeddings {
+    pub fn load(vb: VarBuilder, config: &DebertaV2Config) -> Result<Self> {
+        let device = vb.device().clone();
+        let config = config.clone();
+
+        let embedding_size = config.embedding_size.unwrap_or(config.hidden_size);
+
+        let word_embeddings =
+            embedding(config.vocab_size, embedding_size, vb.pp("word_embeddings"))?;
+
+        let position_embeddings = if config.position_biased_input {
+            Some(embedding(
+                config.max_position_embeddings,
+                embedding_size,
+                vb.pp("position_embeddings"),
+            )?)
+        } else {
+            None
+        };
+
+        let token_type_embeddings: Option<Embedding> = if config.type_vocab_size > 0 {
+            Some(candle_nn::embedding(
+                config.type_vocab_size,
+                config.hidden_size,
+                vb.pp("token_type_embeddings"),
+            )?)
+        } else {
+            None
+        };
+
+        let embed_proj: Option<candle_nn::Linear> = if embedding_size != config.hidden_size {
+            Some(candle_nn::linear_no_bias(
+                embedding_size,
+                config.hidden_size,
+                vb.pp("embed_proj"),
+            )?)
+        } else {
+            None
+        };
+
+        let layer_norm = layer_norm(
+            config.hidden_size,
+            config.layer_norm_eps,
+            vb.pp("LayerNorm"),
+        )?;
+
+        let position_ids =
+            Tensor::arange(0, config.max_position_embeddings as u32, &device)?.unsqueeze(0)?;
+
+        Ok(Self {
+            word_embeddings,
+            position_embeddings,
+            token_type_embeddings,
+            layer_norm,
+            position_ids,
+            device,
+            config,
+            embedding_size,
+            embed_proj,
+        })
+    }
+
+    pub fn forward(
+        &self,
+        input_ids: Option<&Tensor>,
+        token_type_ids: Option<&Tensor>,
+        position_ids: Option<&Tensor>,
```

Comment:
We always have their values (`input_ids`, `token-type_ids`, `position_ids`), so I think `Tensor` type would be enough